File: src/main/java/sk/hudak/prco/utils/JsoupUtils.java
package sk.hudak.prco.utils;

import lombok.NonNull;
import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;

import java.util.Optional;

public class JsoupUtils {

    private JsoupUtils() {
    }

    public static int calculateCountOfPages(int countOfProduct, int pagging) {
        int hh = countOfProduct % pagging;

        int result = countOfProduct / pagging;

        if (hh > 0) {
            return result + 1;
        } else {
            return result;
        }
    }

    public static String hrefAttribute(Element element) {
        if (element == null) {
            return null;
        }
        return element.attr("href");
    }

    public static String srcAttribute(Element element) {
        if (element == null) {
            return null;
        }
        return element.attr("src");
    }

    public static String dataSrcAttribute(Element element) {
        if (element == null) {
            return null;
        }
        return element.attr("data-src");
    }

    public static boolean existElement(@NonNull Document document, @NonNull String cssQuery) {
        return !notExistElement(document, cssQuery);
    }

    public static boolean notExistElement(@NonNull Document document, @NonNull String cssQuery) {
        return document.select(cssQuery).isEmpty();
    }

    public static Optional<Element> getFirstElementByClass(@NonNull Document document, @NonNull String className) {
        return Optional.ofNullable(document.getElementsByClass(className).first());
    }

    public static Optional<String> getTextFromFirstElementByClass(@NonNull Document document, @NonNull String className) {
        Optional<Element> firstElementByClass = getFirstElementByClass(document, className);
        if (!firstElementByClass.isPresent()) {
            return Optional.empty();
        }
        //TODO NPE
        String text = firstElementByClass.get().getAllElements().get(0).text();
        return Optional.of(text);
    }

}

File: src/main/java/sk/hudak/prco/utils/JsoupUtils.kt
package sk.hudak.prco.utils

import org.jsoup.nodes.Document
import org.jsoup.nodes.Element
import org.jsoup.select.Elements
import java.util.*

/**
 * Same as elements.attr("src")
 */
fun Elements.src(): String =
        this.attr("src")

/**
 * Same as element.attr("src")
 */
fun Element.src(): String =
        this.attr("src")


/**
 * Same as elements.attr("href")
 */
fun Elements.href(): String =
        this.attr("href")

/**
 * Same as element.attr("href")
 */
fun Element.href(): String =
        this.attr("href")


object JsoupUtils {

    fun calculateCountOfPages(countOfProducts: Int, pagging: Int): Int {
        val hh = countOfProducts % pagging
        val result = countOfProducts / pagging
        return if (hh > 0) {
            result + 1
        } else {
            result
        }
    }

    fun dataSrcAttribute(element: Element?): String? {
        return element?.attr("data-src")
    }

    fun existElement(document: Document, cssQuery: String): Boolean {
        return !notExistElement(document, cssQuery)
    }

    fun notExistElement(document: Document, cssQuery: String): Boolean {
        return document.select(cssQuery).isEmpty()
    }

    fun getFirstElementByClass(document: Document, className: String): Optional<Element> {
        return Optional.ofNullable(document.getElementsByClass(className).first())
    }

    fun getTextFromFirstElementByClass(document: Document, className: String): Optional<String> {
        val firstElementByClass = getFirstElementByClass(document, className)
        if (!firstElementByClass.isPresent) {
            return Optional.empty()
        }
        //TODO NPE
        val text = firstElementByClass.get().allElements[0].text()
        return Optional.of(text)
    }

}
